### PR TITLE
feat: get organization details by id

### PR DESCRIPTION
**Pull request by Amal-Salam**
### What does the PR do?
This PR addresses issue [#147](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/147)  by implementing the functionality to get organization details by id. This endpoint will provide the details of the organization linked to the id.
 
### How should this be manually tested?
To test this feature, you can send a GET request to the following endpoint:

 ### GET/api/v1/organizations/:org_id
    Authorization: Bearer eyJhbGciOiJIUz

 On a successful products category retrieval, the API should return a 200  status code.
Request Example:
`GET /api/v1/organizations/:org_id`

### Success Response:

         
 ```
{
 "status": 200,
  "message": "Fetched Organization Details Successfully",
  "data": {
    "id": "string",
    "created_at": "datetime",
    "updated_at": "datetime",
    "name": "string",
    "description": "string",
    "email": "string",
    "industry": "string",
    "type": "string",
    "country": "string",
    "address": "string",
    "state": "string",
    "isDeleted": boolean
  }
}
```
GetOrgById-Success

### Error Response:
Bad Request:
```
{
  "status": "unsuccessful",
 "status_code": 400,
  "message": "Must Provide a valid organization Id"
}
```
GetOrgById-BadRequest

Not Found:
```
{
  "status": "unsuccessful",
 "status_code": 404,
  "message": "Organization not found"
}
```

## Requirements
Develop server-side logic to retrieve organization details by ID.
Securely handle the retrieval process and comply with security standards.
Ensure that the organization ID is validated and handled correctly.

### Tasks:

- [x] Define the GET /api/v1/organizations/:org_id route in the server-side application.
- [x] Validate that the organization ID is provided and is in a valid format.
- [x] Implement logic to retrieve the organization details from the database using the provided ID.
- [x] Return the organization details in the response body if the ID is valid and the organization exists.
- [x] Handle cases where the organization ID is invalid or the organization does not exist, providing appropriate error 
   messages.
Reference Issue: https://github.com/hngprojects/hng_boilerplate_nestjs/issues/147
Reference Ticket: https://linear.app/team-bingo/issue/BAC2-72/get-organization-details-or-147

### CODE REVIEW

**Review by Heba-WebDev — APPROVED** [review]

File: src/modules/organisations/organisations.service.ts [hunk 1]
```diff
@@ -123,4 +123,19 @@ export class OrganisationsService {
...
+      const orgDetails = await this.organisationRepository.findOne({ where: { id: orgId } });
+      if (!orgId) {
+        throw new BadRequestException('Must Provide a valid organization Id');
+      }
+      return { message: 'Fetched Organization Details Successfully', data: orgDetails };
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new InternalServerErrorException('An internal server error occured');
```
Comment (Cobidex) [inline src/modules/organisations/organisations.service.ts:138]: try / catch not needed. interceptor will handle that for you

Comment (Amal-Salam) [reply src/modules/organisations/organisations.service.ts]: I have implemented this change.

File: src/modules/organisations/organisations.service.ts [hunk 2]
```diff
@@ -123,4 +124,17 @@ export class OrganisationsService {
       throw new InternalServerErrorException(`An internal server error occurred: ${error.message}`);
     }
   }
+
+  async getOrganizationDetailsById(orgId: string) {
+    if (!isUUID(orgId)) {
+      throw new BadRequestException('Must Provide a valid organization Id');
+    }
```
Comment (davidolanrewaju) [inline src/modules/organisations/organisations.service.ts:131]: Please make use of the global error exception helper in declaring errors

File: src/modules/organisations/organisations.service.ts [hunk 3]
```diff
@@ -123,4 +124,17 @@ export class OrganisationsService {
...
+
+    const orgDetails = await this.organisationRepository.findOne({ where: { id: orgId } });
+
+    if (!orgDetails) {
+      throw new NotFoundException('Organization Id Not Found');
+    }
```
Comment (davidolanrewaju) [inline src/modules/organisations/organisations.service.ts:137]: Same here

File: src/modules/organisations/organisations.service.ts (hunk 2 above)
Comment (Amal-Salam) [reply src/modules/organisations/organisations.service.ts:132]: noted.
i have implemented this.

File: src/modules/organisations/organisations.service.ts (hunk 3 above)
Comment (Amal-Salam) [reply src/modules/organisations/organisations.service.ts:138]: done

**Review by Heba-WebDev — APPROVED** [review]

**Review by davidolanrewaju — DISMISSED** [review]

**Review by davidolanrewaju — DISMISSED** [review]

**Review by Cobidex — APPROVED** [review]

**Pull request merged**